packages/@tailwindcss-upgrade/src/migrate-js-config.ts

```diff
@@ -0,0 +1,121 @@
+    )
+    return
+  }
+
+  let cssConfigs: string[] = []
+
+  if ('content' in unresolvedConfig) {
+    cssConfigs.push(migrateContent(unresolvedConfig as any))
+  }
+
+  if ('theme' in unresolvedConfig) {
+    cssConfigs.push(await migrateTheme(unresolvedConfig as any))
+  }
+
+  console.log(cssConfigs.join('\n'))
+}
+
+async function migrateTheme(unresolvedConfig: Config & { theme: any }): Promise<string> {
+  let { extend: extendTheme, ...overwriteTheme } = unresolvedConfig.theme
+
+  let resetNamespaces = new Set()
+
+  let css = `@theme reference inline {\n`
+  for (let [key, value] of themeableValues(overwriteTheme)) {
+    if (typeof value !== 'string' && typeof value !== 'number') {
+      continue
+    }
+
+    if (!resetNamespaces.has(key[0])) {
+      resetNamespaces.add(key[0])
+      css += `  --${keyPathToCssProperty([key[0]])}-*: initial;\n`
+    }
+
+    css += `  --${keyPathToCssProperty(key)}: ${value};\n`
+  }
+
+  for (let [key, value] of themeableValues(extendTheme)) {
+    if (typeof value !== 'string' && typeof value !== 'number') {
+      continue
+    }
+
+    css += `  --${keyPathToCssProperty(key)}: ${value};\n`
+  }
+
+  return css + '}\n'
+}
+
+function migrateContent(unresolvedConfig: Config & { content: any }): string {
+  let css = ''
+  for (let content of unresolvedConfig.content) {
+    if (typeof content !== 'string') {
+      throw new Error('Unsupported content value: ' + content)
+    }
+
+    css += `@source  "${content}";\n`
+  }
+  return css
+}
+
+// Applies heuristics to determine if we can attempt to migrate the config
+async function isSimpleConfig(unresolvedConfig: Config, source: string): Promise<boolean> {
+  // The file may not contain any functions
+  if (source.includes('function') || source.includes(' => ')) {
+    return false
+  }
+
+  // The file may not contain non-serializable values
+  const isSimpleValue = (value: unknown): boolean => {
+    if (typeof value === 'function') return false
+    if (Array.isArray(value)) return value.every(isSimpleValue)
+    if (typeof value === 'object' && value !== null) {
+      return Object.values(value).every(isSimpleValue)
+    }
+    return ['string', 'number', 'boolean', 'undefined'].includes(typeof value)
+  }
+  if (!isSimpleValue(unresolvedConfig)) {
+    return false
+  }
+
+  // The file may only contain known-migrateable high-level properties
+  const knownProperties = ['content', 'theme', 'plugins', 'presets']
+  if (Object.keys(unresolvedConfig).some((key) => !knownProperties.includes(key))) {
+    return false
+  }
+  if (unresolvedConfig.plugins && unresolvedConfig.plugins.length > 0) {
+    return false
+  }
+  if (unresolvedConfig.presets && unresolvedConfig.presets.length > 0) {
+    return false
+  }
+
+  // The file may not contain any imports
+  if (source.includes('import') || source.includes('require')) {
+    return false
+  }
+
+  return true
+}
```

Comment:
Would be cool if we can chunk declarations into related groups - we'd just need to find a good breakpoint here? I guess a new group is started if the key path except the last part update? 

Other than that we also need to keep track of properties that we could not put into CSS.